### PR TITLE
[5.1]Installation Language

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #36048 .

### Summary of Changes

At present the Language Selection in Joomla Installer displays only the nativeName in the list. This makes language names to display in different places not alphabetically by [English] name. This is true especially for language names that do not contain Latin characters. For example, Spanish would appear immediately after English as the native name for Spanish is Español. You can also notice other language such as Tamil (with native name தமிழ்)is listed after the English alphabet names.

Suggestion is to show the language name (ie English Name) along with native name in the list and the list sorted in the language name. This would help to see the languages alphabetically in English. In addition, it would help to find the language easily.

### Actual result BEFORE applying this Pull Request
image

### Expected result AFTER applying this Pull Request
image

### CODE REVIEW

**Comment by Kostelano** [comment]
It still depends on localization teams. There are several languages where the `<nativeName>` and `<name>` headers are reversed.

Someone needs to contact the localization teams (e.g. Bulgarian, etc.) and fix this. Then we get the result you describe - "alphabetical" and "easy to find".

Just in case, I'll note @tecpromotion. Don't kick me if it's none of your business :).

**Comment by brianteeman** [comment]
> > I raised this a long long time ago so I 10000% agree
> 
> And you pr got rejected? If you create a pr I would support it.

**Comment by brianteeman** [comment]
@HLeithner I made this because you asked

**Comment by HLeithner** [comment]
@brianteeman thanks for bringing this up again, I will check and find some testers hopefully soon.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by brianteeman** [comment]
If this PR is never going to be accepted then please close it. I only made it at the request of the release lead

**Comment by HLeithner** [comment]
alpha2 packages are already ready... @LadySolveig can you merge this at some point in time?

**Comment by richard67** [comment]
The system tests fail at installing. Can be seen here when using the "show all logs" at the top or downloading the complete log. The log then shows:

```
1) Install Joomla
       Install Joomla:
     CypressError: Timed out retrying after 4000ms: `cy.select()` failed because it could not find a single `<option>` with value, index, or text matching: `English (United Kingdom)`
```

It happens in file `install/Installation.cy.js`.

@brianteeman Do you think you can adapt the system tests to your change, or should we try to find someone who can help? Unfortunately I have no knowledge on Cypress.

**Comment by brianteeman** [comment]
The issue is not with the test but with the upstream joomla-cypress package node_modules\joomla-cypress\src\joomla.js

**Comment by richard67** [comment]
> The issue is not with the test but with the upstream joomla-cypress package node_modules\joomla-cypress\src\joomla.js

Yes, I meanwhile know, and @LadySolveig will handle it.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 186eca26a67ced80bf17cb014f9f4254d2c11310<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38573">issues.joomla.org/tracker/joomla-cms/38573</a>.</sub>

File: libraries/src/Language/LanguageHelper.php [hunk 1]
```diff
@@ -61,6 +61,38 @@ public static function createLanguageList($actualLanguage, $basePath = JPATH_BAS
...
+    {
+        $list      = [];
+        $clientId  = $basePath === JPATH_ADMINISTRATOR ? 1 : 0;
+        $languages = $installed ? static::getInstalledLanguages($clientId, true) : self::getKnownLanguages($basePath);
+
+        foreach ($languages as $languageCode => $language) {
+            $metadata = $installed ? $language->metadata : $language;
+
+            $list[] = [
+                'text'     => $metadata['name'] . " | " . $metadata['nativeName'] ?? $metadata['name'],
```
Comment (richard67) [inline libraries/src/Language/LanguageHelper.php]:
```suggestion
                'text'     => $metadata['name'] . ' | ' . $metadata['nativeName'] ?? $metadata['name'],
```

Comment (richard67) [reply libraries/src/Language/LanguageHelper.php]: We should use single quotes whenever possible and double quotes only when necessary or when they simplify things like `"\n"`.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on d42fc0af79403b4f6acedf23e26161dd263e8a6b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38573">issues.joomla.org/tracker/joomla-cms/38573</a>.</sub>

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on d42fc0af79403b4f6acedf23e26161dd263e8a6b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38573">issues.joomla.org/tracker/joomla-cms/38573</a>.</sub>

**Comment by richard67** [comment]
I've set RTC despite of the failing system tests as @LadySolveig told me that she will adjust the tests when this PR gets merged.

**Comment by richard67** [comment]
I just see there is already a PR for adjusting the system tests: https://github.com/joomla-projects/joomla-cypress/pull/11

Thanks @alikon .

**Comment by brianteeman** [comment]
Any reason why this RTC pull request is not being merged?

**Comment by LadySolveig** [comment]
Thank you @brianteeman and for the review and for testing to @richard67 @dautrich

**Comment by brianteeman** [comment]
yay!!

**Pull request merged**